### PR TITLE
docs/design: update template

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve?

Problem Summary: the current design doc template is not detailed enough and lacks directivity. Update the template to make it clearer.

### What is changed and how it works?

What's Changed: design doc template is updated.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`: N/A

### Check List 

Tests 

- No code

Side effects

### Release note

- N/A

### CODE REVIEW

**Comment by zz-jason** [comment]
@bb7133 @XuHuaiyu PTAL

File: docs/design/TEMPLATE.md [hunk 1]
```diff
@@ -8,86 +8,100 @@ This is a template for TiDB's change proposal process, documented [here](./READM
...
+- Compatibility with other external components, like PD, TiKV, TiFlash, BR, TiCDC, Dumpling, TiUP, K8s, etc.
+- Upgrade compatibility
+- Downgrade compatibility
+-->
+
+### Benchmark Tests
+
+<!--
+The following two parts need to be measured:
+- measure the performance of this feature under different parameters
```
Comment (bb7133) [inline docs/design/TEMPLATE.md:85]:
```suggestion
- The performance of this feature under different parameters
```

File: docs/design/TEMPLATE.md [hunk 2]
```diff
@@ -8,86 +8,100 @@ This is a template for TiDB's change proposal process, documented [here](./READM
...
+### Compatibility Tests
 
 <!--
-A detailed description for each step in the implementation:
-- Does any former steps block this step?
-- Who will do it?
-- When to do it?
-- How long it takes to accomplish it?
+A checklist to test compatibility:
+- Compatibility with other features, like partition table, security&privilege, collation&charset, clustered index, async commit, etc.
```
Comment (bb7133) [inline docs/design/TEMPLATE.md:74]:
```suggestion
- Compatibility with other features, like partition table, security & privilege, charset & collation, clustered index, async commit, etc.
```

File: docs/design/TEMPLATE.md [hunk 3]
```diff
@@ -8,86 +8,100 @@ This is a template for TiDB's change proposal process, documented [here](./READM
...
+- Upgrade compatibility
+- Downgrade compatibility
+-->
+
+### Benchmark Tests
+
+<!--
+The following two parts need to be measured:
+- measure the performance of this feature under different parameters
+- measure the performance influence on the online workload
```
Comment (bb7133) [inline docs/design/TEMPLATE.md:86]:
```suggestion
- The performance influence on the online workload
```

File: docs/design/TEMPLATE.md [hunk 4]
```diff
@@ -8,86 +8,100 @@ This is a template for TiDB's change proposal process, documented [here](./READM
...
+- The performance of this feature under different parameters
+- The performance influence on the online workload
+-->
+
+## Impacts & Risks
+
+<!--
+Describe the potential impacts & risks of the design on overall performance, security, k8s, and other aspects. List all the risks or unknowns by far.
+
+Please describe impacts and Risks in two sections: Impacts could be positive or negative, and intentional. Risks are usually negative, unintentional, and may or may not happen. E.g., for performance, we might expect a new feature to improve latency by 10% (expected impact), there is a risk that latency in scenarios X and Y could degrade by 50%.
```
Comment (XuHuaiyu) [inline docs/design/TEMPLATE.md:94]: Risks --> risks

Comment (zz-jason) [reply docs/design/TEMPLATE.md]: resolved.

**Review by XuHuaiyu — APPROVED** [review]
LGTM

**Pull request merged**